File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/HardDrive.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import HardDrive._

case class HardDrive (
  id: Option[Integer],
short_desc: Option[String],
size: Option[String],
drive_type: Option[String],
monthly_price: Option[BigDecimal],
monthly_price_display: Option[String])

object HardDrive {
  import DateTimeCodecs._

  implicit val HardDriveCodecJson: CodecJson[HardDrive] = CodecJson.derive[HardDrive]
  implicit val HardDriveDecoder: EntityDecoder[HardDrive] = jsonOf[HardDrive]
  implicit val HardDriveEncoder: EntityEncoder[HardDrive] = jsonEncoderOf[HardDrive]
}

File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/HardDrive.js
goog.provide('API.Client.HardDrive');

/**
 * A hard drive option available for a dedicated server configuration.
 * @record
 */
API.Client.HardDrive = function() {}

/**
 * @type {!number}
 * @export
 */
API.Client.HardDrive.prototype.id;

/**
 * @type {!string}
 * @export
 */
API.Client.HardDrive.prototype.shortDesc;

/**
 * @type {!string}
 * @export
 */
API.Client.HardDrive.prototype.size;

/**
 * @type {!string}
 * @export
 */
API.Client.HardDrive.prototype.driveType;

/**
 * @type {!number}
 * @export
 */
API.Client.HardDrive.prototype.monthlyPrice;

/**
 * @type {!string}
 * @export
 */
API.Client.HardDrive.prototype.monthlyPriceDisplay;

